### ISSUE TITLE
Categorical encodings on new data when using trained model

### ISSUE THREAD

**Issue by alexHeu** [dmlc/xgboost#8431]
Hi,

I trained a model using the categorical support with pandas/cudf X["cat_feature"].astype("category") and enable_categorical in the DMatrix creation.

When I use the model downstream for creating predictions with X_new["cat_feature"].astype("category") and again constructing a DMatrix, am I responsible to ensure that the same cat_codes are used in the new data or is xgboost handling that internally?

Mentioning this case in the documentation would be helpful.

Thanks!

**Comment by trivialfis** [comment]
Yes, callers need to ensure the consistency of data. Will update the document.

**Comment by alexHeu** [comment]
@trivialfis thanks! Any best practices that you can recommend?

**Comment by trivialfis** [comment]
I think there's nothing special about categorical features, best practices are common to all kinds of features. Make sure your data processing pipeline is consistently applied to all data. For instance, if you have normalized the training data, test data should have the same normalization based on the training data. The same goes for any other techniques you might apply to train data like whitening, and categorical encoding.
